transform/views/main.py

```diff
@@ -54,236 +47,50 @@ def server_error(error=None):
-        return server_error(e)
-    logger.info("IMAGES:SUCCESS")
-
-    return send_file(zipfile.in_memory_zip, mimetype='application/zip', add_etags=False)
-
-
 @app.route('/common-software', methods=['POST'])
 @app.route('/common-software/<sequence_no>', methods=['POST'])
-@app.route('/common-software/<sequence_no>/<batch_number>', methods=['POST'])
-def common_software(sequence_no=1000, batch_number=0):
-    survey_response = request.get_json(force=True)
-
-    if batch_number:
-        batch_number = int(batch_number)
-
-    if sequence_no:
-        sequence_no = int(sequence_no)
-
-    survey = get_survey(survey_response)
-    if not survey:
-        return client_error("CS:Unsupported survey/instrument id")
-
-    survey_id = survey_response.get("survey_id")
-    try:
-        if survey_id == "009":
-            transformer = MBSTransformer(survey_response, sequence_no)
-        elif survey_id == "134":
-            transformer = MWSSTransformer(survey_response, sequence_no, log=logger)
-        else:
-            transformer = CSTransformer(logger, survey, survey_response, batch_number, sequence_no)
-
-        transformer.create_zip()
-
-    except Exception as e:
-        tx_id = survey_response.get("tx_id")
-        logger.exception("CS:could not create files for survey", survey_id=survey_id, tx_id=tx_id)
-        return server_error(e)
-
-    logger.info("CS:SUCCESS")
-
-    return send_file(transformer.image_transformer.get_zip(), mimetype='application/zip', add_etags=False)
+def common_software(sequence_no=1000):
+    return transform(sequence_no)
 
 
 @app.route('/cora', methods=['POST'])
 @app.route('/cora/<sequence_no>', methods=['POST'])
 def cora(sequence_no=1000):
-    survey_response = request.get_json(force=True)
-
-    if sequence_no:
-        sequence_no = int(sequence_no)
-
-    survey = get_survey(survey_response)
-    if not survey:
-        return client_error("CORA:Unsupported survey/instrument id")
-
-    survey_id = survey_response.get("survey_id")
-    try:
-        if survey_id == "144":
-            transformer = UKISTransformer(survey_response, sequence_no)
-            transformer.create_zip()
-        else:
-            return client_error("CORA survey with survey id {} is not supported".format(survey_id))
-
-    except Exception as e:
-        tx_id = survey_response.get("tx_id")
-        logger.exception("CORA:could not create files for survey", survey_id=survey_id, tx_id=tx_id)
-        return server_error(e)
-
-    logger.info("CORA:SUCCESS")
-
-    return send_file(transformer.image_transformer.get_zip(), mimetype='application/zip', add_etags=False)
+    return transform(sequence_no)
 
 
 @app.route('/cord', methods=['POST'])
 @app.route('/cord/<sequence_no>', methods=['POST'])
 def cord(sequence_no=1000):
+    return transform(sequence_no)
+
+
+@app.route('/transform', methods=['POST'])
+@app.route('/transform/<sequence_no>', methods=['POST'])
+def transform(sequence_no=1000):
```

Comment:
Having all these endpoints call the same function seems odd.  A lot of this (I think, I've not tested it) can be rewritten to:
```python
@app.route('/common-software', methods=['POST'])
@app.route('/common-software/<sequence_no>', methods=['POST'])
@app.route('/cora', methods=['POST'])
@app.route('/cora/<sequence_no>', methods=['POST'])
@app.route('/cord', methods=['POST'])
@app.route('/cord/<sequence_no>', methods=['POST'])
@app.route('/transform', methods=['POST'])
@app.route('/transform/<sequence_no>', methods=['POST'])
def transform(sequence_no=1000):
...
```

Even this looks a bit weird to me though and could maybe be a further refactor.  If all endpoints do basically the same thing, then should to flatten all the endpoints into a single one?

Reply:
After this refactor we will be able to remove the logic from downstream where it decides which endpoint to call.
Once that is gone we can remove all of those annotations.

Reply:
Your suggestion does work, I thinks thats best for now.